extension.Dockerfile

```diff
@@ -0,0 +1,51 @@
+# Build the manager binary
+FROM --platform=$BUILDPLATFORM golang:1.22 AS builder
+
+WORKDIR /workspace
+# Copy the Go Modules manifests
+COPY go.mod go.mod
+COPY go.sum go.sum
+# cache deps before building and copying source so that we don't need to re-download as much
+# and so that source changes don't invalidate our downloaded layer
+RUN go mod download
+
+# Copy the go source
+COPY cmd/main.go cmd/main.go
+COPY cmd/extension/main.go cmd/extension/main.go
+COPY api/ api/
+COPY internal/ internal/
+COPY pkg/ pkg/
+
+# Set environment variables for cross-compilation
+ARG TARGETARCH
+
+# Build
+
+ARG GIT_SHA
+ARG DIRTY
+ARG VERSION
+
+ENV GIT_SHA=${GIT_SHA:-unknown}
+ENV DIRTY=${DIRTY:-unknown}
+ENV VERSION=${VERSION:-unknown}
+
+RUN CGO_ENABLED=0 GOOS=linux GOARCH=${TARGETARCH} go build -a -ldflags "-X main.version=${VERSION} -X main.gitSHA=${GIT_SHA} -X main.dirty=${DIRTY}" -o manager cmd/main.go
+
+
+RUN mkdir -p extensions/myextension
+RUN CGO_ENABLED=0 GOOS=linux GOARCH=${TARGETARCH} go build -a -o extensions/myextension/myextension cmd/extension/main.go
+
+# Use distroless as minimal base image to package the manager binary
+# Refer to https://github.com/GoogleContainerTools/distroless for more details
+FROM gcr.io/distroless/static:nonroot
```

Comment:
Had issues with Podman using a local image as a base. I initially wanted to use the normal process to build the image, but depending on the value of `IMG` add our extension executable into it using a builder layer. Would not work in the case of `IMG` set to quay.io/kuadrant/kuadrant-operator:dev as this doesn't exist in quay and wasn't able to pull the image